job.py

```diff
@@ -79,3 +79,9 @@ def write_stage_info(self, query_id, prefix):
 
     f.write("%s\t%s\t%s\n" % (query_id, last_stage_runtime, self.original_runtime()))
     f.close()
+
+  def ideal_time_s(self, num_cores_per_executor):
+    total_time = 0
+    for stage in self.stages.itervalues():
+      total_time += stage.ideal_time_s(num_cores_per_executor)
```

Comment:
I find this more readable! Also I'm not sure there's much value in optimizing things like this in our parsing code.